src/core/device.c

```diff
@@ -697,6 +697,12 @@ static int device_setup_unit(Manager *m, sd_device *dev, const char *path, bool
         if (dev && device_is_bound_by_mounts(DEVICE(u), dev))
                 device_upgrade_mount_deps(u);
 
+        if (ready_units) {
+                r = set_ensure_put(ready_units, NULL, DEVICE(u));
+                if (r < 0)
+                        return log_unit_error_errno(u, r, "Failed to store unit: %m");
+        }
```

Comment:
Should we just call `ready_units` something more generic here, like `set`? We can pass any set to `device_setup_unit()`, so it seems weird to give it such a specific name here.